internal/cli/atlas/deployments/search/indexes/create.go

```diff
@@ -87,10 +112,43 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
 	}
 
 	opts.index, err = db.CreateSearchIndex(ctx, opts.index.CollectionName, opts.index)
+	return err
+}
+
+func (opts *CreateOpts) RunAtlas() error {
+	if err := opts.validateAndPrompt(); err != nil {
+		return err
+	}
+
+	index, err := opts.NewSearchIndex()
 	if err != nil {
 		return err
 	}
-	return nil
+
+	opts.index, err = opts.store.CreateSearchIndexes(opts.ConfigProjectID(), opts.DeploymentName, index)
+	return err
+}
+
+func (opts *CreateOpts) Run(ctx context.Context) error {
+	if opts.DeploymentType == "" {
+		if err := opts.PromptDeploymentType(); err != nil {
+			return err
+		}
+	}
+
+	if strings.EqualFold(opts.DeploymentType, "local") {
+		return opts.RunLocal(ctx)
+	}
+
+	if !opts.IsCliAuthenticated() {
+		return ErrNotAuthenticated
+	}
+
+	if opts.DeploymentName == "" {
+		return ErrNoDeploymentName
+	}
```

Comment:
is this error possible? if their did not provide the deployment name they will get the prompt right?

Reply:
ah, yes not anymore - thanks for catching

Reply:
ah, no, nevermind, we should keep this, for Atlas right? as we don't list atlas clsters yet we ask for user to provide the name 

~/workplace/mongocli CLOUDP-197016 ⇡2 *27 +2 !1 ?2 ❯ atlasdev deployments search index create                                                                                                                                                                 35s  1.20 15:06:31
? What type of deployment would you like to work with? atlas
Error: deployment name is required for Atlas resources